core/server/common/src/main/java/alluxio/RpcUtils.java

```diff
@@ -61,19 +86,51 @@
     try {
       return callable.call();
     } catch (AlluxioException e) {
-      LOG.debug("Internal Alluxio error when running rpc", e);
+      LOG.debug("Internal Error: {}", callable, e);
+      throw e.toThrift();
+    } catch (IOException e) {
+      LOG.debug("I/O Error: {}", callable, e);
+      throw new ThriftIOException(e.getMessage());
+    } catch (Exception e) {
+      LOG.error("Unexpected Error: {}", callable, e);
+      throw new UnexpectedAlluxioException(e).toThrift();
+    }
+  }
+
+  /**
+   * Calls the given {@link RpcCallableThrowsIOException} and handles any exceptions thrown. This
+   * method also logs enter and exit when debug level logging is enabled.
+   *
+   * @param logger the logger to use for this call
+   * @param callable the callable to call
+   * @param <T> the return type of the callable
+   * @return the return value from calling the callable
+   * @throws AlluxioTException if the callable throws an Alluxio or runtime exception
+   * @throws ThriftIOException if the callable throws an IOException
+   */
+  public static <T> T call(Logger logger, RpcCallableThrowsIOException<T> callable)
+      throws AlluxioTException, ThriftIOException {
+    try {
+      logger.debug("Enter: {}", callable);
+      T ret = callable.call();
+      logger.debug("Exit: {}", callable);
+      return ret;
+    } catch (AlluxioException e) {
+      logger.debug("Internal Error: {}", callable, e);
       throw e.toThrift();
     } catch (IOException e) {
-      LOG.warn("I/O error when running rpc", e);
+      logger.debug("I/O Error: {}", callable, e);
       throw new ThriftIOException(e.getMessage());
     } catch (Exception e) {
-      LOG.error("Unexpected error running rpc", e);
+      logger.error("Unexpected Error: {}", callable, e);
       throw new UnexpectedAlluxioException(e).toThrift();
     }
   }
 
   /**
-   * An interface representing a callable which can only throw Alluxio exceptions.
+   * An interface representing a callable which can only throw Alluxio exceptions. The toString
+   * method of classes implementing this interface should be in the format "CallName.
```

Comment:
the `.` in the format string feels odd ... when I see `.`, I intuitively think "end of statement"

how about replacing `.` with `:` or alternatively `CallName(arg1:value1, arg2:value2, ...)`?